src/webnn_native/ops/Gru.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 The WebNN-native Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "webnn_native/ops/Gru.h"
+
+#include <algorithm>
+
+#include "common/Log.h"
+#include "webnn_native/Error.h"
+
+namespace webnn_native { namespace op {
+    Gru::Gru(GraphBuilderBase* builder,
+             OperandBase* input,
+             OperandBase* weight,
+             OperandBase* recurrentWeight,
+             size_t steps,
+             size_t hiddenSize,
+             GruOptions const* options)
+        : OperandBase(builder, {input, weight, recurrentWeight}) {
+        if (options != nullptr) {
+            mOptions = *options;
+            if (options->bias != nullptr) {
+                mInputs.push_back(options->bias);
+            }
+            if (options->recurrentBias != nullptr) {
+                mInputs.push_back(options->recurrentBias);
+            }
+            if (options->initialHiddenState != nullptr) {
+                mInputs.push_back(options->initialHiddenState);
+            } else {
+                mOptions.bias = nullptr;
+                mOptions.recurrentBias = nullptr;
+                mOptions.initialHiddenState = nullptr;
+            }
+        }
+        mSteps = steps;
+        mHiddenSize = hiddenSize;
+        mOptions.resetAfter = options == nullptr ? true : options->resetAfter;
+        mOptions.returnSequence = options == nullptr ? false : options->returnSequence;
+        mOptions.direction =
```

Comment:
I think L33 has set `mOptions.direction` and `mOptions.layout`, no need to add L51~54

Reply:
Optimized.